src/validation.cpp

```diff
@@ -3007,7 +3033,8 @@ static bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state,
     if(!proofSize)
         return state.Invalid(false, REJECT_INVALID, "bad-proof", "No Proof in block");
 
-    CPubKey aggregatePubkey = FederationParams().GetAggPubkeyFromHeight(nHeight);
+    CPubKey aggregatePubkey = aggPubkeys ?
+         [nHeight](const std::vector<aggPubkeyAndHeight>& list) { return list.rbegin()->aggpubkey; }(*aggPubkeys) : FederationParams().GetAggPubkeyFromHeight(nHeight);
 
     if(!aggregatePubkey.IsValid())
```

Comment:
There are following check in the above code, why is this check necessary?

```cpp
if(!aggregatePubkeyinBlock.IsFullyValid())
```

Reply:
This is the check used to validate pubkey inputs throughout the code base like `importpubkey` rpc. It checks the pub key validity using `secp256k1` library.

Reply:
yes, I mean

```cpp
    switch((TAPYRUS_XFIELDTYPES)block.xfieldType)
    {
        case TAPYRUS_XFIELDTYPES::AGGPUBKEY:
            if(block.xfield.size() == CPubKey::COMPRESSED_PUBLIC_KEY_SIZE)
            {
                CPubKey aggregatePubkeyinBlock(block.xfield.begin(), block.xfield.end());
                if(!aggregatePubkeyinBlock.IsFullyValid())
                    return state.Invalid(false, REJECT_INVALID, "bad-aggpubkey", "invalid aggregatePubkey");
            }
            else
                return state.Invalid(false, REJECT_INVALID, "bad-xfieldType-xfield", "invalid aggregatePubkey");
            break;
...
```

This code checks aggregated pubkey validity in `aggregatePubkeyinBlock.IsFullyValid()`, but after that

```cpp
    CPubKey aggregatePubkey = aggPubkeys ?
         [nHeight](const std::vector<aggPubkeyAndHeight>& list) { return list.rbegin()->aggpubkey; }(*aggPubkeys) : FederationParams().GetAggPubkeyFromHeight(nHeight);

    if(!aggregatePubkey.IsValid())
        return state.Error("Invalid aggregatePubkey");
```

this code checks ` if(!aggregatePubkey.IsValid())`. This check is necessary?

Reply:
ok! I didn't understand the question... its fixed now